### PR TITLE
feat(trials): fill Tier-1 field gaps in CTGov and WHO importers

### PR DESCRIPTION
**Pull request by brunoamaral**
## Summary

Wires up six `Trials` columns that the importers were already receiving from upstream but silently dropping — pure importer changes, **no model changes, no migrations**.

| # | Field(s) | Source | Gap |
|---|---|---|---|
| 1 | `results_url_link` | WHO ICTRP XML | Tag present in export, missing from plain-field list |
| 2 | `study_design` | ClinicalTrials.gov API | WHO-only; now composed from `designModule.designInfo` |
| 3 | `results_ipd_plan`, `results_ipd_description` | ClinicalTrials.gov API | WHO-only; now from `ipdSharingStatementModule` |
| 4 | `secondary_sponsor` | ClinicalTrials.gov API | Collaborators were parsed but then discarded |
| 5 | `last_refreshed_on` | ClinicalTrials.gov API | WHO + EUCTIS only; now from `lastUpdatePostDateStruct` |
| 6 | `date_enrollement`, `contact_affiliation` | ClinicalTrials.gov API | WHO-only; enrollment from `startDateStruct`, affiliation from `overallOfficials[0]` |

**Bonus:** `countries` string is now built with `sorted(set(...))` instead of `list(set(...))`, making it deterministic and preventing spurious `HistoricalRecords` rows on repeated pipeline runs with identical data.

## What changed

- **`django/gregory/management/commands/importWHOXML.py`** — added `results_url_link` to the plain-field list; flows through the existing non-destructive update path unchanged.
- **`django/gregory/classes.py`** (`parse_study_to_clinical_trial`) — extracts and exposes the six new keys in `extra_fields`; `study_design` is a deterministic human-readable string (parts joined with `. `); `whoMasked` list is sorted; `results_ipd_plan` truncated to 10 chars to fit `CharField(max_length=10)`; `contact_affiliation` guards against absent/empty `overallOfficials`.
- **`django/gregory/management/commands/feedreader_trials_ctgov.py`** — each new key added in both `create_new_trial` kwargs **and** `extra_field_mapping` in `update_existing_trial` (the two-place rule from the plan; this is exactly the gap that caused `secondary_sponsor` to be lost).

## Update semantics

All new fields use the existing **non-destructive merge**: only overwrite when the incoming value is non-empty, so a missing field never blanks data a previous source populated (matches `docs/trials-multi-source-merge.md`).

## Tests

Three new test modules, 37 new tests — all green:

- `gregory/tests/test_ctgov_tier1_fields.py` — parser unit tests (20 tests): each new field, edge cases (absent module → `None`, partial dates, sorted `whoMasked`, determinism).
- `gregory/tests/management/test_ctgov_tier1_fields.py` — command integration tests (14 tests): create stores each field; update replaces non-empty values but never blanks existing ones.
- `gregory/tests/test_who_importer.py` — WHO importer integration tests (3 tests): `results_url_link` on create, fill-empty update, and non-blank guard.

Full suite: **467/467 pass**. `makemigrations --check --dry-run` confirms no schema changes.

## Test plan

- [ ] `docker exec gregory python manage.py test gregory.tests.test_ctgov_tier1_fields` — 20 tests
- [ ] `docker exec gregory python manage.py test gregory.tests.management.test_ctgov_tier1_fields` — 14 tests
- [ ] `docker exec gregory python manage.py test gregory.tests.test_who_importer` — 3 tests
- [ ] `docker exec gregory python manage.py test gregory.tests` — full suite (467 tests)
- [ ] `docker exec gregory python manage.py makemigrations --check --dry-run` — no changes
- [ ] After merging, run the one-time `backfill_trial_acronyms` command on prod if not yet done (see memory note)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: django/gregory/classes.py [hunk 1]
```diff
@@ -562,6 +562,48 @@ def parse_study_to_clinical_trial(self, study_data: dict) -> 'ClinicalTrial':
...
+			if design_info.get('observationalModel'):
+				parts.append(f"Observational model: {design_info['observationalModel']}")
+			if design_info.get('timePerspective'):
+				parts.append(f"Time perspective: {design_info['timePerspective']}")
+			study_design = '. '.join(parts) if parts else None
+
+		# results_ipd_plan / results_ipd_description
+		ipd_module = protocol.get('ipdSharingStatementModule', {})
+		results_ipd_plan = (ipd_module.get('ipdSharing') or '')[:10] or None
+		results_ipd_description = ipd_module.get('description') or None
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/gregory/classes.py]: Fixed in 9c02f266: strip() before [:10] so truncation reflects real content; whitespace-only description becomes None.

File: django/gregory/classes.py [hunk 2]
```diff
@@ -562,6 +562,48 @@ def parse_study_to_clinical_trial(self, study_data: dict) -> 'ClinicalTrial':
...
+
+		# secondary_sponsor — collaborators are already collected above
+		secondary_sponsor = '; '.join(collaborators) if collaborators else None
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/gregory/classes.py]: Fixed in 9c02f266: collaborator names are stripped and blank entries filtered out before joining, so whitespace-only entries can't produce a spurious non-empty string.

File: django/gregory/classes.py [hunk 3]
```diff
@@ -562,6 +562,48 @@ def parse_study_to_clinical_trial(self, study_data: dict) -> 'ClinicalTrial':
...
+
+		# last_refreshed_on
+		last_refreshed_on = self._parse_date(status_module.get('lastUpdatePostDateStruct', {}).get('date'))
+
+		# date_enrollement — same start date already used for published_date
+		date_enrollement = self._parse_date(start_date_struct.get('date'))
+
+		# contact_affiliation — first overall official's affiliation
+		overall_officials = contacts_module.get('overallOfficials', [])
+		contact_affiliation = overall_officials[0].get('affiliation') if overall_officials else None
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/gregory/classes.py]: Fixed in 9c02f266: strip() applied; whitespace-only affiliation becomes None so it won't pass the update guard.

File: django/gregory/tests/test_who_importer.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	with tempfile.NamedTemporaryFile(mode='w', suffix='.xml', delete=False) as f:
+		f.write(xml_content)
+		path = f.name
+	try:
+		cmd = WHOCommand()
+		cmd.stdout = open(os.devnull, 'w')
+		cmd.parse_xml(path, source_id)
+	finally:
+		os.unlink(path)
+		cmd.stdout.close()
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/gregory/tests/test_who_importer.py]: Fixed in 9c02f266: cmd is now created inside a context manager block for devnull; finally only unlinks the temp file and can't raise UnboundLocalError.
